llvm/lib/Transforms/Utils/FunctionComparator.cpp

```diff
@@ -143,6 +143,14 @@ int FunctionComparator::cmpAttrs(const AttributeList L,
         if (int Res = cmpNumbers((uint64_t)TyL, (uint64_t)TyR))
           return Res;
         continue;
+      } else if (LA.isConstantRangeAttribute() &&
+                 RA.isConstantRangeAttribute()) {
```

Comment:
This still needs the getKindAsEnum check (in case there are more ConstantRange attributes in the future).

Might make sense to just move that check outside the isTypeAttribute branch and always perform it.

Reply:
nice catch. added the check. it needs to be inside the if branch as otherwise string attribute will not work